gems/aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -469,5 +456,26 @@ def determine_profile(options)
       ret ||= 'default'
       ret
     end
+
+    def sso_session(cfg, profile, sso_session_name)
+      sso_session = cfg["sso-session #{sso_session_name}"]
+
+      if sso_session.nil? && sso_session_name.match(/\s/)
```

Comment:
You can possible just add an `|| cfg["sso-session '#{sso_session_name}'"]` after `cfg["sso-session #{sso_session_name}"]`

Reply:
Yeah, we can skip the check for it containing whitespace and always use this check - I think thats reasonable and forwards compatible.